presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -113,13 +114,15 @@ public LogicalPlanner(
             Session session,
             PlanNodeIdAllocator idAllocator,
             Metadata metadata,
-            PlanVariableAllocator variableAllocator)
+            VariableAllocator variableAllocator)
     {
         this.session = requireNonNull(session, "session is null");
         this.idAllocator = requireNonNull(idAllocator, "idAllocator is null");
         this.metadata = requireNonNull(metadata, "metadata is null");
-        this.variableAllocator = requireNonNull(variableAllocator, "variableAllocator is null");
-        this.statisticsAggregationPlanner = new StatisticsAggregationPlanner(variableAllocator, metadata.getFunctionAndTypeManager().getFunctionAndTypeResolver());
+        requireNonNull(variableAllocator, "variableAllocator is null");
+        checkState(variableAllocator instanceof PlanVariableAllocator, "variableAllocator should be instance of PlanVariableAllocator");
```

Comment:
I still don't like this. if the variableAllocator interface isn't working for us can we change it? (clean up PlanVariableAllocator and move to the SPI or improve the VariableAllocator interface so we can ues it straight through the planner/optimizer without having to cast to PlanVariableAllocator?  It seems like this is causing trouble for multiple cases right now.

Reply:
Its a valid point. We do have redundancy with VariableAllocator and PlanVariableAllocator. Which we should clean up. However this would require some thoughts, and I would suggest to take up this is as a separate change. WDYT?

Reply:
discussed offline that we'll add a todo here and clean up VariableAllocator vs. PlanVariableAllocator in a follow up PR.

Reply:
Yeah for this change this is better and more generic approach.